addons/test_mail/tests/test_performance.py

```diff
@@ -1196,21 +1197,21 @@ def test_complete_message_post(self):
         ]
         attachments = self.env['ir.attachment'].with_user(self.env.user).create(self.test_attachments_vals)
         with self.assertQueryCount(employee=49):
-            self.cr.sql_log = self.warm and self.cr.sql_log_count
-            record_container.with_context({}).message_post(
-                body='<p>Test body <img src="cid:cid1"> <img src="cid:cid2"></p>',
-                subject='Test Subject',
-                message_type='notification',
-                subtype_xmlid=None,
-                partner_ids=recipients.ids,
-                parent_id=False,
-                attachments=attachments_vals,
-                attachment_ids=attachments.ids,
-                email_add_signature=True,
-                model_description=False,
-                mail_auto_delete=True
-            )
-            self.cr.sql_log = False
+            enable_logging = self.cr._enable_logging() if self.warm else nullcontext()
+            with enable_logging:
```

Comment:
FWIW this CM could be folded into the previous `with` to avoid needing to reindent everything.